api/v1alpha1/bundle_types.go

```diff
@@ -51,13 +51,27 @@ type BundleSource struct {
 	Type string `json:"type"`
 	// Image is the bundle image that backs the content of this bundle.
 	Image *ImageSource `json:"image,omitempty"`
+	// Git is the git repository that backs the content of this Bundle.
+	Git *GitSource `json:"git,omitempty"`
 }
 
 type ImageSource struct {
 	// Ref contains the reference to a container image containing Bundle contents.
 	Ref string `json:"ref"`
 }
 
+type GitSource struct {
+	Repository string `json:"repository"`
+	Directory  string `json:"directory,omitempty"`
+	Ref        GitRef `json:"ref"`
+}
+
+type GitRef struct {
```

Comment:
follow-up: validate (either in code or schema validation?) that only a single branch/tag/commit has been specified?

Reply:
Yes -- I have it specified in the code and would like to do it via schema validation as well in #173